src/graph.js

```diff
@@ -40,37 +40,91 @@ class Graph {
   // Optionally accepts an array of other GraphNodes for the new vertex to be connected to
   // Returns the newly-added vertex
   addVertex(value, edges = []) {
-
+    const vertex = new GraphNode({ value, edges }); 
+    if (this.vertices.length === 0) {
+      this.vertices.push(vertex);
+      return vertex;
+    }
+    if (this.vertices.length === 1) {
+      this.vertices.push(vertex);
+      this.vertices[0].pushToEdges(vertex);
+      vertex.pushToEdges(this.vertices[0]);
+      return vertex;
+    }
+    this.vertices.push(vertex);
+    for (let i = 0; i < vertex.edges.length; i++) {
+      vertex.edges[i].pushToEdges(vertex);
+    }
+    return vertex;
   }
   // Checks all the vertices of the graph for the target value
   // Returns true or false
   contains(value) {
-
+    for (let i = 0; i < this.vertices.length; i++) {
+      if (this.vertices[i].value === value) return true; 
+    }
+    return false;
   }
   // Checks the graph to see if a GraphNode with the specified value exists in the graph 
   // and removes the vertex if it is found
   removeVertex(value) {
-
+    for (let i = 0; i < this.vertices.length; i++) {
+      if (this.vertices[i].value === value) {
+        this.vertices.splice(i, 1);  
+      }
+    }
   }
   // Checks the two input vertices to see if each one references the other in their respective edges array
   // Both vertices must reference each other for the edge to be considered valid
   // If only one vertex references the other but not vice versa, should not return true
   // Note: You'll need to store references to each vertex's array of edges so that you can use 
   // array methods on said arrays. There is no method to traverse the edge arrays built into the GraphNode class
   checkIfEdgeExists(fromVertex, toVertex) {
```

Comment:
I would definitely try to not have this implemented using nested loops. It really affects performance. Feel free to reference the solution code to see another way of doing it that achieves linear performance.